### PR TITLE
Update FloatExtensions.swift

### PR DESCRIPTION
**Pull request by HanksHu**
Add methods to try four-byte and float data interrotation

### CODE REVIEW

**Comment by HanksHu** [comment]
@guykogus 
A float takes four bytes in a computer。
The property bitPattern of float is of type UInt32, converted to 4 bytes
Sorry，I don't know how to add tests and a changelog😅。
```Swift
//examples:
let num: Float = 27.2435355 // 27.24354
num.bitPattern // This is the system property 1104802499(UInt32)
num.bitPatterns // This is what I added [65, 217, 242, 195] (UInt8)  four bytes
let bits: [UInt8] = [0x41, 0xd9, 0xf2, 0xc3] // [65, 217, 242, 195] four bytes
Float(bitPatterns: bits) // 27.24354
```

File: Sources/SwifterSwift/SwiftStdlib/FloatExtensions.swift [hunk 1]
```diff
@@ -28,7 +28,19 @@ public extension Float {
     var double: Double {
         return Double(self)
     }
-
+    ///The float type takes 4 bytes and returns the corresponding 4 bytes
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftStdlib/FloatExtensions.swift:31]:
```suggestion

    /// The 4 bytes of the `bitpattern` value.
```

File: Sources/SwifterSwift/SwiftStdlib/FloatExtensions.swift [hunk 2]
```diff
@@ -28,7 +28,19 @@ public extension Float {
...
+    }
+    ///Four bytes to float data
+    init(bitPatterns: [UInt8]) {
+        if bitPatterns.count != 4 {
+            self =  0.0
+        } else {
+            let sum = [0, 1, 2, 3].map { (UInt32(bitPatterns[$0]) << (24-$0*8))}.reduce(into: 0, +=)
+            self = .init(bitPattern: sum)
+        }
+    }
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftStdlib/FloatExtensions.swift:43]:
```suggestion

    /// Initialise a `Float` from its bit patterns
    /// - Parameter bitPatterns: The 4 bytes of `Float` data
    init(bitPatterns: [UInt8]) {
        assert(bitPatterns.count == 4)
        self.init(bitPattern: bitPatterns.enumerated().reduce(UInt32(), { $0 + (UInt32($1.element) << (24 - $1.offset * 8)) }))
    }
```

Comment (HanksHu) [reply Sources/SwifterSwift/SwiftStdlib/FloatExtensions.swift:43]: 👍   Thank you very much for your help. I'll study hard. Try it next time. @guykogus

**Pull request closed without merging**